src/opf/acp.jl

```diff
@@ -1,293 +1,167 @@
-        - sum(pf[a] for a in ref[:bus_arcs][i])
-        - sum(shunt["gs"] for shunt in bus_shunts[i])*vm[i]^2
+        sum(gen_status[g] * pg[g] for g in bus_gens[i])
+        - sum(branch_status[e] * pf[e] for e in bus_arcs_fr[i])
+        - sum(branch_status[e] * pt[e] for e in bus_arcs_to[i])
+        - gs[i] * vm[i]^2
         == 
-        sum(load["pd"] for load in bus_loads[i])
+        pd[i]
     )
-    JuMP.@constraint(model,
+    @constraint(model,
         kirchhoff_reactive[i in 1:N],
-        sum(qg[g] for g in ref[:bus_gens][i]) 
-        - sum(qf[a] for a in ref[:bus_arcs][i])
-        + sum(shunt["bs"] for shunt in bus_shunts[i])*vm[i]^2
+        sum(gen_status[g] * qg[g] for g in bus_gens[i])
+        - sum(branch_status[e] * qf[e] for e in bus_arcs_fr[i])
+        - sum(branch_status[e] * qt[e] for e in bus_arcs_to[i])
+        + bs[i] * vm[i]^2
         ==
-        sum(load["qd"] for load in bus_loads[i])
+        qd[i]
     )
 
-    # Branch power flow physics and limit constraints
-    # We pre-allocate constraint containers for simplicity
-    model[:thermal_limit_fr] = Vector{JuMP.ConstraintRef}(undef, E)
-    model[:thermal_limit_to] = Vector{JuMP.ConstraintRef}(undef, E)
-    model[:voltage_difference_limit] = Vector{JuMP.ConstraintRef}(undef, E)
-    model[:ohm_active_fr] = Vector{JuMP.ConstraintRef}(undef, E)
-    model[:ohm_active_to] = Vector{JuMP.ConstraintRef}(undef, E)
-    model[:ohm_reactive_fr] = Vector{JuMP.ConstraintRef}(undef, E)
-    model[:ohm_reactive_to] = Vector{JuMP.ConstraintRef}(undef, E)
-    for (i,branch) in ref[:branch]
-        data["branch"]["$i"]["br_status"] == 0 && continue  # skip branch
-
-        f_idx = (i, branch["f_bus"], branch["t_bus"])
-        t_idx = (i, branch["t_bus"], branch["f_bus"])
-
-        p_fr = pf[f_idx]
-        q_fr = qf[f_idx]
-        p_to = pf[t_idx]
-        q_to = qf[t_idx]
-
-        vm_fr = vm[branch["f_bus"]]
-        vm_to = vm[branch["t_bus"]]
-        va_fr = va[branch["f_bus"]]
-        va_to = va[branch["t_bus"]]
-
-        g, b = PM.calc_branch_y(branch)
-        tr, ti = PM.calc_branch_t(branch)
-        ttm = tr^2 + ti^2
-        g_fr = branch["g_fr"]
-        b_fr = branch["b_fr"]
-        g_to = branch["g_to"]
-        b_to = branch["b_to"]
-
-        # From side of the branch flow
-        model[:ohm_active_fr][i] = JuMP.@constraint(model, 
-              ((g+g_fr)/ttm) * vm_fr^2
-            + ((-g*tr+b*ti)/ttm) * (vm_fr*vm_to*cos(va_fr-va_to)) 
-            + ((-b*tr-g*ti)/ttm) * (vm_fr*vm_to*sin(va_fr-va_to)) 
-            - p_fr 
-            ==
-            0
+    model[:sm_fr] = Vector{ConstraintRef}(undef, E)
+    model[:sm_to] = Vector{ConstraintRef}(undef, E)
+    model[:va_diff] = Vector{ConstraintRef}(undef, E)
+    model[:ohm_pf] = Vector{ConstraintRef}(undef, E)
+    model[:ohm_qf] = Vector{ConstraintRef}(undef, E)
+    model[:ohm_pt] = Vector{ConstraintRef}(undef, E)
+    model[:ohm_qt] = Vector{ConstraintRef}(undef, E)
```

Comment:
Same as SOC-OPF: we can vectorize this. Hopefully won't trigger another round of CI errors 🤞

Reply:
See https://github.com/AI4OPT/OPFGenerator/pull/119/commits/47d10c0d450211ec40a3f1884aa708ca32e67a0f

Reply:
CI errors on SOC formulation but not AC 🎉